mp/src/game/shared/momentum/mom_stickybomb.cpp

```diff
@@ -0,0 +1,387 @@
+        pStickybomb->SetAbsAngles(angles);
+
+        pStickybomb->SetDamage(120.0f);
+        pStickybomb->SetRadius(146.0f);
+    }
+
+    return pStickybomb;
+}
+
+void CMomStickybomb::InitStickybomb(const Vector &velocity, const AngularImpulse &angVelocity,
+                                    CBaseCombatCharacter *pOwner)
+{
+    SetupInitialTransmittedGrenadeVelocity(velocity);
+
+    IPhysicsObject *pPhysicsObject = VPhysicsGetObject();
+    if (pPhysicsObject)
+    {
+        pPhysicsObject->AddVelocity(&velocity, &angVelocity);
+    }
+}
+
+void CMomStickybomb::RemoveStickybomb(bool bNoGrenadeZone)
+{
+    // Kill it
+    SetThink(&BaseClass::SUB_Remove);
+    SetNextThink(gpGlobals->curtime);
+    SetTouch(nullptr);
+    AddEffects(EF_NODRAW);
+
+    if (bNoGrenadeZone)
+    {
+        // Sprite flash
+        CSprite *pGlowSprite = CSprite::SpriteCreate(NOGRENADE_SPRITE, GetAbsOrigin(), false);
+        if (pGlowSprite)
+        {
+            pGlowSprite->SetTransparency(kRenderGlow, 255, 255, 255, 255, kRenderFxFadeFast);
+            pGlowSprite->SetThink(&CSprite::SUB_Remove);
+            pGlowSprite->SetNextThink(gpGlobals->curtime + 1.0);
+        }
+    }
+}
+
+void CMomStickybomb::Detonate()
+{
+    trace_t tr;
+    Vector vecSpot;
+
+    SetThink(nullptr);
+
+    vecSpot = GetAbsOrigin() + Vector(0, 0, 8);
+    UTIL_TraceLine(vecSpot, vecSpot + Vector(0, 0, -32), MASK_SHOT_HULL, this, COLLISION_GROUP_NONE, &tr);
+
+    Explode(&tr, tr.m_pEnt);
+
+    if (CNoGrenadesZone::IsInsideNoGrenadesZone(this))
+    {
+        RemoveStickybomb(true);
+        return;
+    }
+
+    if (m_bFizzle)
+    {
+        g_pEffects->Sparks(GetAbsOrigin());
+        RemoveStickybomb(true);
+        return;
+    }
+}
+
+void CMomStickybomb::Explode(trace_t *pTrace, CBaseEntity *pOther)
+{
+    if (CNoGrenadesZone::IsInsideNoGrenadesZone(this))
+    {
+        RemoveStickybomb(true);
+        return;
+    }
+
+    // Make invisible
+    SetModelName(NULL_STRING);
+    SetSolid(SOLID_NONE);
+    m_takedamage = DAMAGE_NO;
+
+    // Pull out a bit
+    if (!CloseEnough(pTrace->fraction, 1.0f))
+    {
+        SetAbsOrigin(pTrace->endpos + (pTrace->plane.normal * 1.0f));
+    }
+
+    CBaseEntity *pOwner = GetOwnerEntity();
+
+    float flDamage = GetDamage();
+    float flRadius = GetRadius();
+
+    // Explosion effect on client
+    Vector vecOrigin = GetAbsOrigin();
+    CPVSFilter filter(vecOrigin);
+
+    const Vector vecNormal = m_bUseImpactNormal ? m_vecImpactNormal : pTrace->plane.normal;
+    TE_TFExplosion(filter, vecOrigin, vecNormal, WEAPON_STICKYLAUNCHER);
+
+    Vector vecReported = GetThrower() ? GetThrower()->GetAbsOrigin() : vec3_origin;
+
+    // Damage
+    CTakeDamageInfo info(this, pOwner, vec3_origin, vecOrigin, flDamage, GetDamageType(), 0, &vecReported);
+    RadiusDamage(info, vecOrigin, flRadius, CLASS_NONE, nullptr);
+
+    if (pOther && !pOther->IsPlayer())
+    {
+        UTIL_DecalTrace(pTrace, "Scorch");
```

Comment:
You may want to consider copying the way the `RocketScorch` decal is created/used but for `StickyScorch`